### PR TITLE
Stop auto-scroll at page boundaries

### PR DESCRIPTION
**Pull request by aidenybai**
## Motivation

Auto-scroll kept requesting animation frames while the pointer stayed near an edge even after the page reached its scroll boundary. Each frame attempted the same no-op scroll indefinitely.

## Example

While dragging upward at the top of the page, `window.scrollBy(0, -8)` leaves `scrollY` at `0`, but React Grab previously scheduled another frame forever.

## Changes

- Continue the auto-scroll loop only when the page actually moved.
- Keep the existing pointer-move restart behavior.
- Add tests for both boundary stopping and continued scrolling.

## Validation

- `nr build`
- `nr --filter react-grab test:unit` — 161 passed
- `nr lint`
- `nr typecheck`
- `nr format`

---
## Summary by cubic
Stop auto-scroll in `react-grab` at page edges to avoid endless `requestAnimationFrame` loops and no-op scrolls. The loop now re-schedules only when a frame actually changes `scrollX`/`scrollY`, calls `onScrollStep` only on real movement, and uses `window.scrollBy({ behavior: "instant" })` for deterministic steps; pointer-move restarts still work, and unit tests cover boundary stopping, continued scrolling, and cancel on stop.

<sup>Written for commit 99c8a45301302ca9083a78f32e5006ab7be3f9bd. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/aidenybai/react-grab/pull/583?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

---

> [!NOTE]
> **Low Risk**
> Small behavior change in drag auto-scroll with clear boundary handling and new unit tests; no auth, data, or API surface changes.
> 
> **Overview**
> Fixes **edge auto-scroll** in React Grab so it no longer spins an endless `requestAnimationFrame` loop when the pointer stays in the edge zone but the page cannot move further (e.g. `scrollY` already at `0` while dragging upward).
> 
> The scroller now tracks whether the last `scrollBy` actually changed `scrollX`/`scrollY` and **only schedules the next frame when a real scroll happened**, instead of continuing whenever the cursor is still near an edge.
> 
> Adds unit tests for stopping at boundaries and for keeping the loop active while scrolling still works.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 8fe2df90739cf78e7db47924dc3cafde7e78672f. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: packages/react-grab/src/core/auto-scroll.ts [hunk 1]
```diff
@@ -48,18 +48,20 @@ export const createAutoScroller = (
...
       const didScrollByY = window.scrollY - previousScrollY;
-      if (didScrollByX !== 0 || didScrollByY !== 0) {
+      didScroll = didScrollByX !== 0 || didScrollByY !== 0;
+      if (didScroll) {
         onScrollStep?.({ x: didScrollByX, y: didScrollByY });
       }
     }
 
-    if (direction.top || direction.bottom || direction.left || direction.right) {
+    if (didScroll) {
```
[the remark this answers is not in the document]
Comment (aidenybai) [reply packages/react-grab/src/core/auto-scroll.ts:68]: Fixed in f70ed872 by using an explicit instant scroll operation, so page CSS cannot defer the position update used by the boundary check. The updated unit stub now exercises the options-based API.
